src/librustc_typeck/collect.rs

```diff
@@ -2256,6 +2267,175 @@ fn predicates_from_bound<'tcx>(
     }
 }
 
+/// Returns the minimum required target-feature name to use a SIMD type on C FFI:
+fn simd_ffi_min_target_feature(
+    target: &str, simd_width: usize, simd_elem_width: usize
+) -> Option<&'static str> {
+    // FIXME: this needs to be architecture dependent and
+    // should probably belong somewhere else:
+    // * on mips: 16 => msa,
+    // * wasm: 16 => simd128
+    match target {
+        t if t.contains("x86") => {
```

Comment:
Would it be possible to deduplicate this block up here with the block down below?